src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/DefaultRazorFormattingService.cs

```diff
@@ -85,6 +85,36 @@ public override async Task<TextEdit[]> FormatAsync(Uri uri, RazorCodeDocument co
             return edits;
         }
 
+        public override Task<TextEdit[]> ApplyFormattedEditsAsync(Uri uri, RazorCodeDocument codeDocument, RazorLanguageKind kind, TextEdit[] formattedEdits, FormattingOptions options)
+        {
+            var span = TextSpan.FromBounds(0, codeDocument.Source.Length);
+            var range = span.AsRange(codeDocument.GetSourceText());
+            var formattingContext = FormattingContext.Create(uri, codeDocument, range, options);
+
+            // TODO
+            var edits = new List<TextEdit>();
+            for (var i = 0; i < formattedEdits.Length; i++)
+            {
+                var projectedRange = formattedEdits[i].Range;
+                if (codeDocument.IsUnsupported() ||
```

Comment:
If the code document is unsupported wont we discard every range? Not sure the value in iterating over all of this

Reply:
Updated it. I am currently returning an empty list for unsupported documents. Should it instead be just return projected edits without any mapping? This comment seems to indicate so but I'm not sure if it makes sense,
`// All language queries on unsupported documents return Html. This is equivalent to what pre-VSCode Razor was capable of.`

Reply:
I think empty edits is fine to be honest